src/types.ts

```diff
@@ -36,6 +36,22 @@ export interface DeviceState {
     color?: AnyColor;
     tilt?: number;
     position?: number;
+    lastUpdate: number;
+    click?: 'single' | 'double' | 'triple' | 'quadruple' | 'long' | 'long_release' | 'many';
+    consumption?: number;
```

Comment:
I think this properties shouldn't be in type, since they are coming from feature.property which is string.
for example if device has feature.name = 'consumption' and feature.property = 'consumption_main' this type wouldn't work

Reply:
We have to stick to `[k: string]: string | number | boolean | OTAState | AnyColor | undefined;` notation

Reply:
I've refactored types. consider rebasing your branch `git rebase master`

Reply:
okay, let settle this down, how we can determine what kind of device it is? if thermostat / button / socket etc?

Reply:
You can't determine device type, only features (using the name property).

Reply:
I will check how `Homebridge Zigbee2MQTT` it's working...  e.g. you want to show device control in the dashboard or kind off... not just exposes as you have in the device setting
![image](https://user-images.githubusercontent.com/1714370/106431412-81649e80-646d-11eb-86fb-e97d85cbfd0b.png)

or have you any better idea? e.g. for climate, we using local_temperature, running_state, and  current_heating_setpoint for rendering the thermostat widget

Reply:
For climate devices, you might have a list of these features.

Reply:
https://github.com/Koenkk/zigbee2mqtt/blob/0f1aad6b98d17a3d2dddc56948f9db7aa72b0c7f/lib/extension/homeassistant.js#L80